src/testing/vortex/workload.zig

```diff
@@ -184,31 +206,104 @@ fn reconcile(result: Result, command: *const Command, model: *Model) !void {
                 try testing.expect(model.account_exists(transfer.debit_account_id));
                 try testing.expect(model.account_exists(transfer.credit_account_id));
             }
+
+            model.latest_transfers.append(successful_transfer_ids.const_slice());
+            model.transfers_created += transfers.len;
         },
         .lookup_all_accounts => |accounts_found| {
-            // Get all known account ids (sorted).
+            // Get all known account ids.
             var id_buffer: [accounts_count_max]u128 = undefined;
             const account_ids_known = model.account_ids(id_buffer[0..]);
 
+            // Check that timestamps are monotonically increasing.
+            var timestamp_max: u64 = 0;
+            for (accounts_found) |account| {
+                if (account.timestamp <= timestamp_max) {
+                    log.err(
+                        "account {d} timestamp {d} is not greater than previous timestamp {d}",
+                        .{ account.id, account.timestamp, timestamp_max },
+                    );
+                    return error.TestFailed;
+                }
+                timestamp_max = account.timestamp;
+            }
+
             // Extract and sort all found account ids.
             var account_ids_found_buffer: [accounts_count_max]u128 = undefined;
             for (accounts_found, 0..) |account, i| {
                 account_ids_found_buffer[i] = account.id;
             }
             const account_ids_found = account_ids_found_buffer[0..accounts_found.len];
-            std.mem.sort(u128, account_ids_found, {}, std.sort.asc(u128));
 
             // All known accounts are found by the query, and no others.
             try testing.expectEqualSlices(u128, account_ids_known, account_ids_found);
 
             try testing.expectEqual(0, debits_credits_difference(accounts_found));
         },
+        .lookup_latest_transfers => |transfers_found| {
+            // Check that timestamps are monotonically increasing.
+            var timestamp_max: u64 = 0;
+            for (transfers_found) |transfer| {
+                if (transfer.timestamp <= timestamp_max) {
+                    log.err(
+                        "transfer {d} timestamp {d} is not greater than previous timestamp {d}",
+                        .{ transfer.id, transfer.timestamp, timestamp_max },
+                    );
+                    return error.TestFailed;
+                }
+                timestamp_max = transfer.timestamp;
+            }
+        },
     }
 }
 
+/// A bounded array of transfer IDs, which appends at the end, dropping items at the start when
+/// necessary. Like a constrained /deque/.
+const LatestsTransfers = struct {
```

Comment:
Can we use RingBuffer from `ring_buffer.zig` here? That won't give you a `.slice`, but I an iterator would work?